---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# germseq

<!-- badges: start -->
<!-- badges: end -->

The goal of germseq is to ...

## Installation

To install the latest version of the package:

``` r
require("devtools")
devtools::install_github("karenliu5/germseq", build_vignettes = TRUE)
library("germseq")
```

## References
Gloor, G., Fernandes, A., Macklaim, J., Albert, A., Links, M., Quinn, T., Wu, J.R.,
Wong, R.G., and B. Lieng (2013). ALDEx2: Analysis Of Differential Abundance Taking
Sample Variation Into Account. R package version 1.30.0.
[https://bioconductor.org/packages/release/bioc/html/ALDEx2.html](https://bioconductor.org/packages/release/bioc/html/ALDEx2.html).

Huang, L. (2020). ANCOMBC: Microbiome differential abudance and correlation
analyses with bias correction. R package version 2.0.1.
[https://www.bioconductor.org/packages/release/bioc/html/ANCOMBC.html](https://bioconductor.org/packages/release/bioc/html/ALDEx2.html).

Lahti, L. and S. Shetty (2019). microbiome: Microbiome Analytics. R package version
1.20.0. [https://www.bioconductor.org/packages/release/bioc/html/microbiome.html](https://www.bioconductor.org/packages/release/bioc/html/microbiome.html).

Lahti, L., Shetty, S., Ernst, F.M. et al. (2021). Orchestrating Microbiome Analysis
with Bioconductor. [microbiome.github.io/oma/](microbiome.github.io/oma/).

Love, M., Huber, W., and S. Anders (2014). DESeq2: Differential gene expression
analysis based on the negative binomial distribution. R package version 1.38.0.
[https://bioconductor.org/packages/release/bioc/html/DESeq2.html](https://bioconductor.org/packages/release/bioc/html/DESeq2.html).

McMurdle, P.J. and S. Holmes (2013). phyloseq: Handling and analysis of high-throughput
microbiome census data. R package version 1.42.0.
[https://www.bioconductor.org/packages/release/bioc/html/phyloseq.html](https://www.bioconductor.org/packages/release/bioc/html/phyloseq.html).

Nearing, J.T., Douglas, G.M., Hayes, M.G., et al. (2022). Microbiome differential abundance methods produce different results across 38 datasets. *Nature Comm.* 13(342). doi:[https://doi.org/10.1038/s41467-022-28034-z](https://doi.org/10.1038/s41467-022-28034-z). 

Olberding, N. (2019). Introduction to the Statistical Analysis of Microbiome Data in R.
[https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/](https://www.nicholas-ollberding.com/post/introduction-to-the-statistical-analysis-of-microbiome-data-in-r/).


## Acknowledgements
This package was developed as part of an assessment for 2022 BCB410H: Applied Bioinformat-
ics course at the University of Toronto, Toronto, CANADA. <PackageName> welcomes issues,
enhancement requests, and other contributions. To submit an issue, use the GitHub issues.


## Example

This is a basic example which shows you how to solve a common problem:

```{r example}
library(germseq)
## basic example code
```

What is special about using `README.Rmd` instead of just `README.md`? You can include R chunks like so:

```{r cars}
summary(cars)
```

You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this. You could also use GitHub Actions to re-render `README.Rmd` every time you push. An example workflow can be found here: <https://github.com/r-lib/actions/tree/v1/examples>.

You can also embed plots, for example:

```{r pressure, echo = FALSE}
plot(pressure)
```

In that case, don't forget to commit and push the resulting figure files, so they display on GitHub and CRAN.
